File: src/main/java/vip/yazilim/libs/springcore/util/ListHelper.java
package vip.yazilim.libs.springcore.util;

import java.util.Collections;
import java.util.List;

/**
 * 
 * @author Emre Sen - Dec 26, 2019
 * @contact maemresen@yazilim.vip
 *
 */
public class ListHelper {

    /**
     * To get a not-null safe list
     * 
     * @param list input list
     * @param <T> type of list items
     * @return not-null list
     */
    public static <T> List<T> getSafeList(List<T> list) {
        if (list == null) {
            return Collections.emptyList();
        }
        return list;
    }

}

File: src/main/java/vip/yazilim/libs/springvip/util/ListHelper.kt
package vip.yazilim.libs.springvip.util

/**
 * To get a not-null safe list
 *
 * @param list input list
 * @param <T> type of list items
 * @return not-null list
</T> */
fun <T> getSafeList(list: List<T?>?): List<T?> {
    return list ?: emptyList()
}
